cloudformation/scylla.yaml

```diff
@@ -0,0 +1,591 @@
+
+Parameters:
+  ClusterName:
+    Type: 'String'
+
+  InstanceCount:
+    Description: Number of EC2 instances (must be between 1 and 10).
+    Type: Number
+    Default: 1
+    MinValue: 1
+    MaxValue: 10
+    ConstraintDescription: Must be a number between 1 and 10.
+
+  ScyllaVersion:
+    Type: String
+    Default: ScyllaDB310rc3
+    AllowedValues:
+      - ScyllaDB310rc3
+
+  InstanceType:
+    Description: EC2 instance type
+    Type: String
+    Default: i3.large
+    AllowedValues:
+      - i3.large
+      - i3.xlarge
+      - i3.2xlarge
+      - i3.4xlarge
+      - i3.8xlarge
+      - i3.16xlarge
+      - i3en.large
+      - i3en.xlarge
+      - i3en.2xlarge
+      - i3en.3xlarge
+      - i3en.6xlarge
+      - i3en.12xlarge
+      - i3en.24xlarge
+      - i3en.metal
+    ConstraintDescription: must be a valid EC2 instance type.
+
+  AvailabilityZone:
+    Type: 'AWS::EC2::AvailabilityZone::Name'
+    ConstraintDescription: must be the name of available AvailabilityZone.
+
+  KeyName:
+    Description: Name of an existing EC2 KeyPair to enable SSH access to the instances
+    Type: 'AWS::EC2::KeyPair::KeyName'
+    ConstraintDescription: must be the name of an existing EC2 KeyPair.
+
+Mappings:
+  AWSRegionVersion2AMI:
+    us-east-1:
+      ScyllaDB310rc3: ami-046e87cc62fb6af2f
+    us-east-2:
+      ScyllaDB310rc3: ami-0e6b225ac591468f9
+    us-west-1:
+      ScyllaDB310rc3: ami-0c6d7f4b7efa5973a
+    us-west-2:
+      ScyllaDB310rc3: ami-01159cfbcce1508fa
+    eu-west-1:
+      ScyllaDB310rc3: ami-03fb02353a3295214
+    eu-west-2:
+      ScyllaDB310rc3: ami-0c7d92f56d768ae64
+    eu-west-3:
+      ScyllaDB310rc3: ami-034c210ab26d91f54
+    eu-north-1:
+      ScyllaDB310rc3: ami-0ac905cc6c6bd8789
+    ap-south-1:
+      ScyllaDB310rc3: ami-08cae15894a13430d
+
+Conditions:
+  Launch1: !Equals [1, 1]
+  Launch2: !Not [!Equals [1, !Ref InstanceCount]]
+  Launch3: !And
+  - !Not [!Equals [1, !Ref InstanceCount]]
+  - !Not [!Equals [2, !Ref InstanceCount]]
+  Launch4: !And
+  - !Not [!Equals [1, !Ref InstanceCount]]
+  - !Not [!Equals [2, !Ref InstanceCount]]
+  - !Not [!Equals [3, !Ref InstanceCount]]
+  Launch5: !And
+  - !Not [!Equals [1, !Ref InstanceCount]]
+  - !Not [!Equals [2, !Ref InstanceCount]]
+  - !Not [!Equals [3, !Ref InstanceCount]]
+  - !Not [!Equals [4, !Ref InstanceCount]]
+  Launch6: !Or
+  - !Equals [6, !Ref InstanceCount]
+  - !Equals [7, !Ref InstanceCount]
+  - !Equals [8, !Ref InstanceCount]
+  - !Equals [9, !Ref InstanceCount]
+  - !Equals [10, !Ref InstanceCount]
+  Launch7: !Or
+  - !Equals [7, !Ref InstanceCount]
+  - !Equals [8, !Ref InstanceCount]
+  - !Equals [9, !Ref InstanceCount]
+  - !Equals [10, !Ref InstanceCount]
+  Launch8: !Or
+  - !Equals [8, !Ref InstanceCount]
+  - !Equals [9, !Ref InstanceCount]
+  - !Equals [10, !Ref InstanceCount]
+  Launch9: !Or
+  - !Equals [9, !Ref InstanceCount]
+  - !Equals [10, !Ref InstanceCount]
+  Launch10: !Equals [10, !Ref InstanceCount]
```

Comment:
Can you please explain the logic of this part?

Reply:
this whole logic cause of cloudformation is limited, and we are hardcoding the nodes we'll create here.
this logic is for selecting which node would be created

Reply:
Worthwhile adding some in template doc

Reply:
done